zipline/data/data_portal.py

```diff
@@ -139,6 +139,9 @@ def __init__(self,
         }
         self._equity_daily_reader_array_data = {}
 
+    def _reindex_extra_source(self, df, source_date_index):
```

Comment:
why did you move this to be a separate method?  (it's not wrong, I'm just curious).  it took me a few reads to realize that it's not a functional change.